dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMain.java

```diff
@@ -67,7 +67,8 @@ public static void main(String[] arguments)
         List<ArtifactProblem> artifactProblems = new ArrayList<>();
         // classPathResult is kept null if JAR files are specified in the argument
         ClassPathResult classPathResult = null;
-    
+        ClassPathBuilder classPathBuilder = null;
```

Comment:
using these in two variables in only one branch of the if-else block is a bit of a code smell. I wonder if there's an alternative approach that doesn't use nullness as flags?

Reply:
It now relies on `artifactsInArguments` variable. It's empty when we specify JAR files.

Reply:
This is becoming more complex, not less. I think the fundamental issue is that two very different cases are being managed with lots of if-else instead of different methods or objects. When I looked at this code in my IDE, it was quite hard to follow.

Reply:
let me try to split the 2 (3?) cases into methods.